src/Symfony/Contracts/HttpClient/ResponseInterface.php

```diff
@@ -22,29 +22,40 @@
  *
  * @author Nicolas Grekas <p@tchwork.com>
  *
- * @throws RedirectionExceptionInterface On a 3xx when getStatusCode() has not been called
- * @throws ClientExceptionInterface      On a 4xx when getStatusCode() has not been called
- * @throws ServerExceptionInterface      On a 5xx when getStatusCode() has not been called
- * @throws TransportExceptionInterface   when a network error occurs
+ * @throws RedirectionExceptionInterface On a 3xx when $throw is true
+ * @throws ClientExceptionInterface      On a 4xx when $throw is true
+ * @throws ServerExceptionInterface      On a 5xx when $throw is true
```

Comment:
should be moved to methods which have this `$throw` argument

Reply:
done